src/SFA.DAS.Commitments.Api.IntegrationTests/DatabaseSetup/Generators/Generator.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using SFA.DAS.Commitments.Api.IntegrationTests.Helpers;
+
+namespace SFA.DAS.Commitments.Api.IntegrationTests.DatabaseSetup.Generators
+{
+    // prefer composition?
+    public class Generator
+    {
+        protected readonly Random Random = new Random();
+
+        public long[] RandomIdGroups(long firstId, int countOfIds, int maxIdsPerGroup, int maxIdsRequired)
+        {
+            var newApprenticeshipIdsShuffled = Enumerable
+                .Range((int)firstId, countOfIds)
+                .OrderBy(au => Random.Next());
+
+            // 'lock in' the enumeration by converting to an array, otherwise you get different results for each enumeration!
+            return newApprenticeshipIdsShuffled
+                .SelectMany(id => Enumerable.Repeat((long)id, Random.Next(1, maxIdsPerGroup + 1)))
+                .Take(maxIdsRequired)
+                .ToArray();
+        }
+
+        //any need for base anymore? yes, if replace above method consumer with this one
+        public T[] RandomIdGroups<T>(long firstId, int countOfIds, ProbabilityDistribution<int> probabilityDistribution,
+            Func<long,int,IEnumerable<T>> generateGroup)
+        {
+            var newApprenticeshipIdsShuffled = Enumerable
+                .Range((int)firstId, countOfIds)
```

Comment:
"could" throw if firstId > int.MaxValue

Reply:
Added an Assert check to check if the id supplied is greater than the method can handle.